pandas/core/internals.py

```diff
@@ -1807,16 +1807,20 @@ def _try_operate(self, values):
         return values.view('i8')
 
     def _try_coerce_args(self, values, other):
-        """ provide coercion to our input arguments
-            we are going to compare vs i8, so coerce to integer
-            values is always ndarra like, other may not be """
+        """ Coerce values and other to dtype 'i8'. NaN and NaT convert to
+            the smallest i8, and will correctly round-trip to NaT if converted
+            back in _try_coerce_result. values is always ndarray-like, other
+            may not be """
         values = values.view('i8')
+
```

Comment:
I'm not sure I follow. We already know the source and destination dtypes: datetime64 -> int64 or timedelta64 -> float64. What do we need to infer?